crates/task-impls/src/consensus.rs

```diff
@@ -590,6 +591,14 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                     return;
                 };
                 let parent_commitment = parent.commit();
+                let Ok(state) = consensus.get_decided_state().validate_and_apply_header(
+                    &proposal.data.block_header.clone(),
+                    &parent.block_header.clone(),
+                    &view,
+                ) else {
+                    error!("Block header doesn't extend the proposal",);
+                    return;
+                };
```

Comment:
Why do we validate against the last decided state instead of the parent state?

Reply:
Good catch! I mistakenly used the last decided state in a few places and I thought I had fixed all of them. Apparently, I missed this one here.